packages/teams-js/test/private/messageChannels.spec.ts

```diff
@@ -0,0 +1,85 @@
+import * as communication from '../../src/internal/communication';
+import { errorLibraryNotInitialized } from '../../src/internal/constants';
+import { messageChannels } from '../../src/private/messageChannels';
+import { app } from '../../src/public/app';
+import { _minRuntimeConfigToUninitialize } from '../../src/public/runtime';
+import { Utils } from '../utils';
+
+describe('messageChannels', () => {
+  // Use to send a mock message from the app.
+  const utils = new Utils();
+
+  beforeEach(() => {
+    utils.processMessage = null;
+    utils.messages = [];
+    utils.childMessages = [];
+    utils.childWindow.closed = false;
+
+    // Mock out MessagePort to support constructor and instanceof checks
+    class MockMessagePort {}
+    global.MessagePort = MockMessagePort as unknown as typeof MessagePort;
+  });
+
+  afterEach(() => {
+    // Reset the object since it's a singleton
+    if (app._uninitialize) {
+      utils.setRuntimeConfig(_minRuntimeConfigToUninitialize);
+
+      app._uninitialize();
+    }
+    // Clear the cached telemetry port
+    messageChannels._clearTelemetryPort();
+  });
+
+  describe('Testing messageChannels.getTelemetryPort', () => {
+    it('should throw if called before initialization', async () => {
+      expect.assertions(1);
+      utils.uninitializeRuntimeConfig();
+      await expect(messageChannels.getTelemetryPort()).rejects.toThrowError(new Error(errorLibraryNotInitialized));
+    });
+
+    it('should return port from message and then from local variable', async () => {
+      expect.assertions(2);
+      await utils.initializeWithContext('content');
```

Comment:
Can you please add other contexts as well and there respective results?

Reply:
This API is not restricted to any particular contexts and will return the same result in all cases.

It didn't seem that the other tests I referenced were testing different contexts unless the behavior differed - is the expected test pattern to test all contexts to demonstrate that the same result is returned?

Reply:
@kerrynf I definitely believe we haven't always been good about testing every context in some of these cases, but in general it's good to test them all because if you were to come back and restrict this function to a smaller set of contexts in the future it would be a breaking change for developers who depend on this function, which we would want to catch in the test layer.

Reply:
Sounds good. :)